pkg/generate/app/sourcelookup.go

```diff
@@ -597,7 +597,11 @@ func CloneAndCheckoutSources(repo git.Repository, remote, ref, localDir, context
 		}
 	}
 	if len(contextDir) > 0 {
-		glog.V(5).Infof("Using context directory %q. The full source path is %q", contextDir, filepath.Join(localDir, contextDir))
+		if _, err := os.Stat(filepath.Join(localDir, contextDir)); os.IsNotExist(err) {
+			return "", fmt.Errorf("supplied context directory '%s' does not exist in '%s' repository", contextDir, remote)
+		} else {
+			glog.V(5).Infof("Using context directory %q. The full source path is %q", contextDir, filepath.Join(localDir, contextDir))
+		}
```

Comment:
Hmm, if it's not caught here, it will give the Dockerfile not found error even if the context directory is what is missing.